### DISCUSSION TITLE
Ref<X, string> or null

### DISCUSSION THREAD

**Discussion by luzzardik** [typegoose/typegoose#918] [Q&A]
Hi,
Either Typescript or Typegoose complains when I want to use a Ref and allow a null value at the same time.
What's the good way to achieve this ?
My field `user` for example should be either a `Ref<User, string>` or `null`.

**Comment by luzzardik** [comment]
I've seen #625, I use `Ref<User,string> | null` for now, but I have the warnings for Mixed. Is it mandatory to use the @prop allowMixed ?

**Reply by hasezoey** [reply]
depends, did you follow [Reference other Classes](https://typegoose.github.io/typegoose/docs/guides/advanced/reference-other-classes)?
if yes, please provide your whole property (or even class), otherwise i cannot tell what would be wrong

**Reply by luzzardik** [reply]
@hasezoey hi !
here's an example from my code
`@prop({default: null, allowMixed: Severity.ALLOW})
    revoked_at!: number | null;`
 and another one:
 `@prop({ref: User, default: null, allowMixed: Severity.ALLOW})
    used_by!: Ref<User, string> | null;`

**Reply by hasezoey** [reply]
`allowMixed` should never be used unless you really actually need it, in your case your problems are:

`@prop({default: null, allowMixed: Severity.ALLOW}) revoked_at!: number | null;`
you set your type to `number | null`, and typescript will output type as `Object` instead of `Number`, so you need to explicitly set [`type`](https://typegoose.github.io/typegoose/docs/api/decorators/prop/#type) to `Number`.

`@prop({ref: User, default: null, allowMixed: Severity.ALLOW}) used_by!: Ref<User, string> | null;`
when `ref` is in use and a different `_id` type than `ObjectId` needs to be used [`type`](https://typegoose.github.io/typegoose/docs/api/decorators/prop/#type)) also always needs to be set.

Note: `type` also needs to always be set for array types, see [`@prop` Array options](https://typegoose.github.io/typegoose/docs/api/decorators/prop#array-options)

TL;DR: if you ever see a Mixed warning, then either typescript didnt output a good type (there are heavy limitations) and you need to manually set the `type` option, or a wrong `type` option has been set
